app/main/controllers/activity/RTMediaBuddyPressActivity.php

```diff
@@ -259,6 +259,12 @@ function bp_get_activity_parent_content( $content ) {
 	 */
 	public function bp_activity_content_before_save( $content ) {
 
+		// Fix duplication issue of comment media on activity by returning contents when activity type is comment.
+		$rtmedia_comment_nonce = filter_input( INPUT_POST, 'rtmedia_comment_nonce', FILTER_SANITIZE_STRING );
+		if ( ! empty( $rtmedia_comment_nonce ) ) {
```

Comment:
Why are we just checking the nonce if it is empty or not instead of verifying? Any reason? Am I missing anything?

cc: @dharmin @surajkrsingh

Reply:
@sanketio 
The code below the condition should need to run if it's not commented. So need to check that the activity which is created is from comment, and the nonce was only feasible field to check if this block is run after posting the comment.

@surajkrsingh Please correct me if something I wrote is incorrect.